src/Robo/Commands/Tests/DrupalTestCommand.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace Acquia\Blt\Robo\Commands\Tests;
+
+use Acquia\Blt\Robo\Exceptions\BltException;
+use Robo\Contract\VerbosityThresholdInterface;
+
+/**
+ * Defines commands in the "tests" namespace.
+ */
+class DrupalTestCommand extends TestsCommandBase {
+
+  const APACHE_RUN_GROUP = 'APACHE_RUN_GROUP';
+  const APACHE_RUN_USER = 'APACHE_RUN_USER';
+  const BROWSERTEST_OUTPUT_DIRECTORY = 'BROWSERTEST_OUTPUT_DIRECTORY';
+  const MINK_DRIVER_ARGS = 'MINK_DRIVER_ARGS';
+  const MINK_DRIVER_ARGS_PHANTOMJS = 'MINK_DRIVER_ARGS_PHANTOMJS';
+  const MINK_DRIVER_ARGS_WEBDRIVER = 'MINK_DRIVER_ARGS_WEBDRIVER';
+  const MINK_DRIVER_CLASS = 'MINK_DRIVER_CLASS';
+  const SIMPLETEST_BASE_URL = 'SIMPLETEST_BASE_URL';
+  const SIMPLETEST_DB = 'SIMPLETEST_DB';
+  const SYMFONY_DEPRECATIONS_HELPER = 'SYMFONY_DEPRECATIONS_HELPER';
+
+  /**
+   * Directory to store output printer files.
+   *
+   * @var string
+   */
+  protected $browsertestOutputDirectory;
+
+  /**
+   * @var string
+   */
+  protected $apacheRunUser;
+
+  /**
+   * @var bool
+   */
+  protected $sudoRunTests;
+
+  /**
+   * Environment varialbes to set for Drupal tests.
+   *
+   * @var array
+   */
+  protected $testingEnv;
+
+  /**
+   * Environment varialbes to exported before Drupal tests.
+   *
+   * @var string
+   */
+  protected $testingEnvString;
+
+  /**
+   * Directory in which test logs and reports are generated.
+   *
+   * @var string
+   */
+  protected $reportsDir;
+
+  /**
+   * The filename for PHPUnit report.
+   *
+   * @var string
+   */
+  protected $reportFile;
+
+  /**
+   * The method for running Drupal test; either phpunit or run-tests.sh.
+   *
+   * @var string
+   */
+  protected $drupalTestRunner;
+
+  /**
+   * This hook will fire for all commands in this command file.
+   *
+   * @hook init
+   */
+  public function initialize() {
+    parent::initialize();
+    $this->reportsDir = $this->getConfigValue('tests.reports.localDir') . '/phpunit';
+    $this->reportFile = $this->reportsDir . '/results.xml';
+    $this->drupalTestRunner = $this->getConfigValue('tests.drupal.test-runner');
+  }
+
+  /**
+   * Setup and run tests.
+   */
+  public function run() {
+    try {
+      $this->setTestingConfig();
+      $this->getTestingEnvString();
+      $this->createLogs();
+      $this->launchWebDriver();
+      $this->executeTests();
+      $this->killWebDriver();
+    }
+    catch (\Exception $e) {
+      // Kill web driver server to prevent Pipelines from hanging after fail.
+      $this->killWebDriver();
+      throw $e;
+    }
+  }
+
+  /**
+   * Executes all Drupal tests in either tests.phpunit or tests.run-tests.
```

Comment:
I'd like to be more clear about what a "Drupal test" is.

Reply:
Updated a bit.